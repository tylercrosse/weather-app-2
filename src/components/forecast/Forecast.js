import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import DayTiles from "./DayTiles";
import Chart from "./Chart";
import "./forecast.css";

export const Forecast = ({ currentTime, timezone, dailyData, hourlyData }) =>
  <section className="forecast">
    <div className="forecast__container">
      <DayTiles timezone={timezone} dailyData={dailyData} />
      <Chart
        currentTime={currentTime}
        timezone={timezone}
        dailyData={dailyData}
        hourlyData={hourlyData}
      />
    </div>
  </section>;

Forecast.propTypes = {
  hourlyData: PropTypes.array.isRequired,
  dailyData: PropTypes.array.isRequired,
  timezone: PropTypes.string.isRequired,
  currentTime: PropTypes.number.isRequired
};

export const mapStateToProps = state => ({
  timezone: state.weather.timezone,
  hourlyData: state.weather.hourly.data,
  dailyData: state.weather.daily.data,
  currentTime: state.weather.currently.time
});

export default connect(mapStateToProps)(Forecast);
