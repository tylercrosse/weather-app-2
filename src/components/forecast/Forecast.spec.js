import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import { Forecast } from './Forecast';

const setup = () => {
  const props = {
    hourlyData: [],
    dailyData: [],
    timezone: 'America/Los_Angeles',
    currentTime: 1499105000
  }

  const component = <Forecast {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    wrapper
  }
}

describe('<Forecast />', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
