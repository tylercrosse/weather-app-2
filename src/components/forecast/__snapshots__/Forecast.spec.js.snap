// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forecast /> should render correctly 1`] = `
<section
  className="forecast"
>
  <div
    className="forecast__container"
  >
    <DayTiles
      dailyData={Array []}
      timezone="America/Los_Angeles"
    />
    <Chart
      currentTime={1499105000}
      dailyData={Array []}
      hourlyData={Array []}
      timezone="America/Los_Angeles"
    />
  </div>
</section>
`;
